frontend/src/components/OrderPage.js

```diff
@@ -113,6 +122,26 @@ class OrderPage extends Component {
     this.getOrderDetails(this.state.orderId);
   }
 
+  handleWebLN = (data) => {
+    console.log(data)
+    getWebLN()
+      .then((webln) => {
+        // If WebLN implements locked payments compatibility, this logic won't be necesary anymore
+        if (data.is_maker & (data.status == 0)) {
+          webln.sendPayment(data.bond_invoice);
+          this.setState({ waitingWebNL: true, openWebNLDialog: true});
+        } else if (data.is_taker & data.status == 3) {
+          webln.sendPayment(data.bond_invoice);
+          this.setState({ waitingWebNL: true, openWebNLDialog: true});
+        } else if (data.is_seller & (data.status == 6 || data.status == 7 )) {
+          webln.sendPayment(data.escrow_invoice);
+          this.setState({ waitingWebNL: true, openWebNLDialog: true});
+        } else {
+          this.setState({ waitingWebNL: false });
+        }
+      });
```

Comment:
This is the only way I found to manage `status` changes without using Hooks, happy to hear any other suggestions I might have missed out.

Reply:
Please also confirm these are the right combination of `user rol + status`